### DISCUSSION TITLE
Extracting the best quality with large datasets.

### DISCUSSION THREAD

**Discussion by chrissype** [autogluon/autogluon#5050] [General]
I have a very large dataset of related series. 15-minutely samples, over 1000 items and multiple years of data. The total number of datapoints is ~90,000,000. At inference time I have access to two weeks of samples, and need to train a single global model to forecast values of these series.

I feel like I should be able to extract very good performance from such a dataset however I have not so far managed it.

Given the constraints of the environment in which the model will be run, fast inference and low disk usage is of very high importance. This has led me to focus on the DirectTabular model as a target, it seems that boosted trees are still competitive in real world forecasting competitions and will fulfil the above criteria.

Can you offer any pointers on whether I'm configuring autogluon correctly for such a task? I noticed some confusing behaviour on this journey in that when I select `"best_quality"` at the `TimeSeriesPredictor` level, this preference is not passed to the `TabularPredictor` fit call, and the data is subsampled. Furthermore, when I increase the amount of data available to the model the number of child models and consequently the disk usage appears to rise. Is that expected?

Here is roughly the config I've settled on so far:

```
    excluded_models = [
        "RecursiveTabular",
        "Theta",
        "SeasonalNaive",
        "Naive",
        "NPTS",
        "DynamicOptimizedTheta",
        "AutoETS",
        "ETS",
        "ChronosZeroShot",
        "ChronosFineTuned",
        "TemporalFusionTransformer",
        "PatchTST",
        "TiDE",
        "ChronosZeroShot[bolt_base]",
        "ChronosFineTuned[bolt_small]",
        "DeepAR",
    ]

    predictor = TimeSeriesPredictor(
        prediction_length=prediction_length,
        target=target_column,
        eval_metric="WQL",
        quantile_levels=[0.1, 0.5, 0.9],
        path=str(save_path),
        log_to_file=True,
        log_file_path="auto",
        freq=train_data.freq,
        # We never really expect to run the same predictions twice
        cache_predictions=False,
    )
    frequency_lags = lags.get_lags_for_frequency(train_data.freq, num_default_lags=16)
    double_lags = [l * 2 for l in frequency_lags if l * 2 < 1344]
    all_lags = sorted(list(set(double_lags + frequency_lags)))
    predictor.fit(
        train_data=train_data,
        presets="best_quality",
        time_limit=time_limit,
        excluded_model_types=excluded_models,
        random_seed=234_000,
        hyperparameters={
            "DirectTabularModel": {
                "max_num_items": None,
                "max_num_samples": 10_000_000,
                "lags": all_lags,
                "date_features": [
                    time_features.minute_of_hour,
                    time_features.hour_of_day,
                    time_features.day_of_week,
                    time_features.day_of_month,
                    time_features.day_of_year,
                    time_features.month_of_year,
                    time_features.week_of_year,
                ],
                "differences": None,
                "target_scaler": "min_max",
                "tabular_fit_kwargs": {
                    "ag_args_fit": {"ag.max_memory_usage_ratio": 2.0},
                    "presets": ["high_quality", "optimize_for_deployment"],
                    "use_bag_holdout": True,
                },
            }
        },
        verbosity=4,
        refit_full=True,
        num_val_windows=10,
        refit_every_n_windows=2,
    )
    predictor.persist("all")
```

I could clearly up the number of data samples, however this already takes 2 days to train on my available compute (70 core 500gb memory server).

Any advice would be greatly appreciated!

**Comment by shchur** [comment]
Hi @chrissype, thank you for your question. Unfortunately, it's hard to tell in advance which model will work best for a specific use case, but there is some general advice that I can share. 

If you want to train a single model (`DirectTabular`), you can set `num_val_windows=1`. AutoGluon uses multiple validation windows only for model selection / ensembling - so if you already know that you want to use a single model, this will be a waste of training time. It only makes sense to increase this parameter if you are training multiple models and want to minimize the risk of overfitting / picking the wrong model.

If you manually provide the `hyperparameters` dictionary, there is no need to also specify the `excluded_model_types` - this argument will have no effect. 

Finally, I would also recommend trying out a fine-tuned version of the Chronos-Bolt model. The model offers very fast inference even on CPU and has potential to show good accuracy given the large amount of training data available in your use case. 
```python
predictor.fit(
    train_data,
    hyperparameters={
        "Chronos": {"model_path": "bolt_base", "fine_tune": True}
    },
    num_val_windows=1,
    refit_full=False,
)
```
You can also try increasing the number of iterations and changing other hyperparameters controlling the fine-tuning process (https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-model-zoo.html#autogluon.timeseries.models.ChronosModel).

Other than that, your setup looks reasonable and I don't think I can provide further guidance here.

**Reply by chrissype** [reply]
Thank you for your advice, that is very helpful!
